src/modules/storage/Storage.ts

```diff
@@ -0,0 +1,71 @@
+import { StorageEventData, StorageEventType, StorageType } from './types';
+
+class StorageClass {
+  private storageListeners: {
+    [StorageType.LOCAL]: ((event: StorageEventData) => void) | null;
+    [StorageType.SESSION]: ((event: StorageEventData) => void) | null;
+  } = {
+    [StorageType.LOCAL]: null,
+    [StorageType.SESSION]: null,
+  };
+
+  initStorageCapture(captureInitialDump = true, localStorage = false, sessionStorage = false): void {
+    if (localStorage || sessionStorage) {
+      if (captureInitialDump) {
+        if (localStorage) this.captureInitialStorageDump(StorageType.LOCAL);
+        if (sessionStorage) this.captureInitialStorageDump(StorageType.SESSION);
+      }
+      window.addEventListener('storage', this.captureStorageEvent);
+    }
+  }
+
+  stopStorageCapture(): void {
+    window.removeEventListener('storage', this.captureStorageEvent);
+  }
+
+  addListener(listener: (event: StorageEventData) => void, storageType: StorageType): void {
+    this.storageListeners[storageType] = listener;
+  }
```

Comment:
```suggestion
  addListener(listener: (event: StorageEventData) => void, storageType: StorageType): void {
    this.storageListeners[storageType] = listener;
    this.initStorageCapture(storageType);
  }
```

Lets combine the two. We can rename the`addListener` method to `startCapturingStorage`.